hoomd/md/CylinderManifold.cc

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2009-2020 The Regents of the University of Michigan
+// This file is part of the HOOMD-blue project, released under the BSD 3-Clause License.
+
+
+// Maintainer: pschoenhoefer
+
+#include "CylinderManifold.h"
+
+namespace py = pybind11;
+
+using namespace std;
+
+/*! \file CylinderManifold.cc
+    \brief Contains code for the CylinderManifold class
+*/
+
+/*!
+    \param r radius of the sphere
+    \param P position of the sphere
+*/
+CylinderManifold::CylinderManifold(std::shared_ptr<SystemDefinition> sysdef,
+                               Scalar r,
+                               Scalar3 P)
+  : Manifold(sysdef), m_r(r) 
+       {
+    m_exec_conf->msg->notice(5) << "Constructing CylinderManifold" << endl;
+    m_surf = manifold_enum::cylinder;
+    validate();
+       }
+
+CylinderManifold::~CylinderManifold() 
+       {
+    m_exec_conf->msg->notice(5) << "Destroying CylinderManifold" << endl;
+       }
+
+        //! Return the value of the implicit surface function of the sphere.
+        /*! \param point The position to evaluate the function.
+        */
+Scalar CylinderManifold::implicit_function(Scalar3 point)
+       {
+       Scalar3 delta = point - m_P;
```

Comment:
We could implement these manifold classes entirely in the `.h` file so that the compiler can inline the computations.